### PR TITLE
Instructions & build support for NTF

### PR DESCRIPTION
**Pull request by nevelis**
Small note on how to compile & activate NTF within OMEC.

### CODE REVIEW

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

**Comment by krsna1729** [comment]
it wont go here. that will be the Makefile

File: INSTALL.md [hunk 1]
```diff
@@ -89,3 +89,14 @@ To drop into BESS shell
 ```bash
 docker exec -it bess bessctl
 ```
+
+## Network Token Functions
+
```
Comment (yiannisy) [inline INSTALL.md:94]: I believe nobody will know about network tokens, let's add some context here?

> OMEC includes a Network Token Function (NTF) which provides preliminary support for Network Tokens, a new protocol to expose datapath services to end users and application providers. More details are available at [https://networktokens.org](https://networktokens.org)

**Comment by yiannisy** [comment]
@krsna1729 do you want to have these instructions on the Makefile and spgwu as comments (instead of install.md)?

**Comment by krsna1729** [comment]
@yiannisy the docker_setup.sh changes was what i was referring to be in Makefile. Since that is where the build happens. Something like this?

```
ENABLE_NTF               ?=

DOCKER_BUILD_ARGS        ?= --build-arg MAKEFLAGS=-j$(shell nproc) --build-arg CPU --build-arg BUILDKIT_INLINE_CACHE
DOCKER_BUILD_ARGS        += --build-arg ENABLE_NTF
```

user can run do
```
ENABLE_NTF=1 ./docker_setup.sh
```
or
```
make docker-build ENABLE_NTF=1
```

**Comment by yiannisy** [comment]
I don't have a build environment to try and test, so I will defer to @nevelis for this :)

File: install_ntf.sh [hunk 2]
```diff
@@ -18,10 +18,11 @@ install_ntf() {
 		doxygen \
 		libjansson-dev \
 		libjansson4 \
-		libtool
+		libtool \
+		unzip
```
Comment (krsna1729) [inline install_ntf.sh:22]: can remove the need for unzip like shown here

https://github.com/omec-project/upf-epc/blob/e471cef3161e716834510477a6fa61844ae8afb8/Dockerfile#L16-L29

Comment (nevelis) [reply install_ntf.sh]: Done!

File: INSTALL.md [hunk 3]
```diff
@@ -89,3 +89,18 @@ To drop into BESS shell
...
+OMEC includes a Network Token Function (NTF) which provides preliminary support
+for Network Tokens, a new protocol to expose datapath services to end users and
+application providers. More details are available at https://networktokens.org
+
+In order to compile NTF support, run the following:
+
+```bash
+USE_NTF=1 ./docker_setup.sh
```
Comment (krsna1729) [inline INSTALL.md:102]:
```suggestion
ENABLE_NTF=1 ./docker_setup.sh
```

File: Makefile [hunk 4]
```diff
@@ -16,6 +20,9 @@ DOCKER_IMAGENAME         := ${DOCKER_REGISTRY}${DOCKER_REPOSITORY}${PROJECT_NAME
 DOCKER_BUILDKIT          ?= 1
 BUILDKIT_INLINE_CACHE    ?= 1
 DOCKER_BUILD_ARGS        ?= --build-arg MAKEFLAGS=-j$(shell nproc) --build-arg CPU --build-arg BUILDKIT_INLINE_CACHE
+ifneq ($(strip $(ENABLE_NTF)),)
+DOCKER_BUILD_ARGS        += --build-arg ENABLE_NTF=$(ENABLE_NTF)
+endif
```
Comment (krsna1729) [inline Makefile:25]: can we remove if guard? with line 13 being 0 by default?

File: Makefile [hunk 5]
```diff
@@ -8,6 +8,10 @@ VERSION                  ?= $(shell cat ./VERSION)
 # For a more portable image set CPU=haswell
 CPU                      ?= native
 
+# Enable Network Token Function support (see https://networktokens.org for more
+# information)
+ENABLE_NTF               ?=
```
Comment (krsna1729) [inline Makefile:13]: this can be

```suggestion
ENABLE_NTF               ?= 0
```

**Comment by krsna1729** [comment]
retest this please

**Comment by nevelis** [comment]
Hey @krsna1729 I committed your USE_NTF => ENABLE_NTF suggestion after your last review, any chance you could check it off?  Thanks!

**Comment by krsna1729** [comment]
@nevelis `retest this please` is the keyword for the jenkins CI :) Jenkins was down, blocking the merge

**Comment by krsna1729** [comment]
retest this please

**Comment by sindack** [comment]
There were some inconsistencies with the `CORD Jenkins - FOSSA Verification`, so I retested this manually (to avoid retriggering everything again) and it passed, see: [https://jenkins.opencord.org/job/omec_upf-epc_fossa/280/](https://jenkins.opencord.org/job/omec_upf-epc_fossa/280/)

**Comment by ajamshed** [comment]
Hi @nevelis. I am sorry... but can you please rebase? I will merge this PR today.

**Comment by nevelis** [comment]
retest this please

**Comment by krsna1729** [comment]
retest this please

**Pull request merged**